CMakeLists.txt

```diff
@@ -946,10 +946,6 @@ set(SWIFT_MAIN_INCLUDE_DIR "${SWIFT_SOURCE_DIR}/include")
 set(SWIFT_SHIMS_INCLUDE_DIR "${SWIFT_SOURCE_DIR}/stdlib/public/SwiftShims")
 set(SWIFT_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/include")
 
-if (NOT BOOTSTRAPPING_MODE)
-  message(FATAL_ERROR "turning off bootstrapping is not supported anymore")
-endif()
-
```

Comment:
I'm against removing this check.

Reply:
Leaving it as a default to `hosttools` is fine. The compiler itself still needs to produce valid output with swift sources disabled, even if it is less optimized. Given that the off-mode also disables macro support, it seems likely that it will be noticed.

Reply:
What about letting the compiler print a warning if SwiftCompilerSources are disabled?

Reply:
>  The compiler itself still needs to produce valid output with swift sources disabled

Sure, but there shouldn't be another motivation to disable SwiftCompilerSources than for cross-compiling where no host-tools are available. And for that case we already do this in cmake file:

```
set(SWIFT_EXEC_FOR_SWIFT_MODULES "${CMAKE_Swift_COMPILER}")
  if(NOT SWIFT_EXEC_FOR_SWIFT_MODULES)
    message(WARNING "BOOTSTRAPPING set to OFF because no Swift compiler is defined")
    set(BOOTSTRAPPING_MODE "OFF")
  endif()
```

Therefore I don't understand why we have to remove this check

Reply:
Right now, there are different flags controlling disabling swift-syntax in the compiler (which drops macros support, among other things) vs. disabling SwiftCompilerSources in the compiler (which drops embedded support and breaks no-locks/no-allocation checking, among other optimizer effects).

I'm a bit dubious on the notion that we have to work hard to prevent people from using bootstrapping=off and being surprised. If it really is that important to prevent misconfigurations, we could create a new CMake + build-script option for this "minimal C++-only build" that disables both of those at once. Checking `SWIFT_EXEC_FOR_SWIFT_MODULES` isn't that great, because it doesn't let you build this configuration if there's a Swift in your path.

Reply:
Yeah, it would be good to have a single flag which turns off everything (macros, swiftcompilersource, etc.). This would make accidental misconfiguration very unlikely. Plus a compiler warning as a bonus.

Reply:
Alright, I've added `build-script --minimal-cxx-bootstrap` to do this, which maps down to the CMake option `SWIFT_MINIMAL_CXX_BOOTSTRAP` and centralizes most of the logic for turning off stuff in the build.